src/applications/vaos/express-care/redux/actions.js

```diff
@@ -145,14 +145,35 @@ export function routeToPageInFlow(flow, history, current, action) {
       type: FORM_PAGE_CHANGE_STARTED,
     });
 
-    const nextAction = flow[current][action];
+    const state = getState();
     let nextPage;
 
-    if (typeof nextAction === 'string') {
-      nextPage = flow[nextAction];
+    if (action === 'next') {
+      const nextAction = flow[current][action];
+
+      if (typeof nextAction === 'string') {
+        nextPage = flow[nextAction];
+      } else {
+        const nextStateKey = await nextAction(getState(), dispatch);
+        if (
+          Object.keys(state.expressCare.newRequest.previousPages).length <= 0
+        ) {
+          Object.assign(state.expressCare.newRequest.previousPages, {
```

Comment:
`Object.assign` mutates the previousPages object here, which breaks the Redux rules. The Redux-y way to do this is to pass the page keys you need into the `FORM_PAGE_CHANGE_COMPLETED` action and update the object in the reducer.

Reply:
Ah yeah the redux rules. OK I can move to the ruducer.

Reply:
Moved to reducer.